packages/cml/src/replaceCmlPlaceholders/replaceCmlPlaceholders.ts

```diff
@@ -78,6 +78,16 @@ const replacePropValues = (mdx: string, values: Record<string, string> = {}): st
       case 'number':
         searchValue = match;
         newValue = `{${propValue}}`;
+        break;
+      case 'json':
+        searchValue = match;
+
+        if (propValue && typeof propValue === 'object') {
```

Comment:
is this needed?

Reply:
why will we pass a non-object value to this?

Reply:
I don't know, fallback like a textarea editor?

Reply:
I think it doesn't hurt to double-check it in order to avoid a case like this where parsing returns a string because it was already a string that had been stringified. 

<img width="410" alt="Screen Shot 2023-07-11 at 19 29 35" src="https://github.com/cmpsr/composer/assets/1816129/9d155b09-e612-481b-a3c1-e4e4026b8727">

Reply:
this will just move the crash to a different place, if a component expect an object and you pass a string or a number the component will crash

Reply:
ok. will remove it

Reply:
Maybe it is not clear, but this component can receive an object or a string. If it receives an object, it stringifies it and then saves it. If it receives a string it will save it as it is.

That's why we have that check.

Reply:
you are thinking on your current use case that I guess you have adapt it to work with both a string or an object, this is a generic feature that has to work for all use cases, if a component expect a json we cannot send a string because that will crash

Reply:
Isn't CML a string that is later parsed? If I send an object, will it be evaluated as a string setting it as `[object Object]`?

Should we stringify it so that when the CML is parsed a proper object is set as prop?

Reply:
I remembered I had tested passing the JSON directly as an object. But this was the result: `[object Object]` . That's why I stringify it.

<img width="729" alt="Screen Shot 2023-07-12 at 10 49 37" src="https://github.com/cmpsr/composer/assets/1816129/fece40d6-952f-4d97-b75d-3c43da13b84a">

Reply:
[The DataViz component expects a JSON array as prop](https://github.com/impulsum/hamlet/pull/154/files#diff-eac2931c9ec5b8f85863b0611ca3511c3414443eb3ca21b38e84a1a6f38b3d1dR4) and it works with this stringify implementation [for instance](https://hamlet-git-ham-345-impulsum-vc.vercel.app/ca/san-mateo-county/burlingame#school-rankings).


I believe that this implementation is correct for all use cases. The JSON is stringified as part of the CML and then parsed when the MDX is generated. 

Because in case it is not stringified, it will be set as `[object Object]` since it is concatenated as a string within the CML by default. And that will cause a crash.

That's my understanding.
Let me know if you have another solution that works.

Reply:
in other words, you only need to do this independently of the type of the data

```
newValue = `{${JSON.stringify(propValue)}}`;
```

Reply:
I don't think so, because if `propValue` is already a string, like in the case of the text area for instance, we end up stringifying a string, which will cause the final component to receive a string instead of an object.

<img width="337" alt="Screen Shot 2023-07-12 at 11 16 33" src="https://github.com/cmpsr/composer/assets/1816129/77380b56-60e6-4ae4-bf0e-16b38058aff0">

Reply:
But let me test it though. If it works we can go with that.

Reply:
- [Generated new canary builds with the proposed solution](https://github.com/cmpsr/composer/pull/755#issuecomment-1632162830)
- [Updated DataViz to use new canary builds with the proposed solution](https://github.com/impulsum/hamlet/pull/154/commits/e19908613b22bf734a8a42267aae163dccb9ccb6)
- Waiting the deploy to finish so I can test it on [contentful entry](https://app.contentful.com/spaces/njp6314ud8bc/environments/master/entries/3BTmeTWWsYmvDrbJOLfs1k) and [preview URL](https://hamlet-git-ham-345-impulsum-vc.vercel.app/ca/san-mateo-county/burlingame#school-rankings)

Reply:
- Updated JSON

<img width="1437" alt="Screen Shot 2023-07-12 at 11 35 55" src="https://github.com/cmpsr/composer/assets/1816129/9be9db44-a562-45a9-8f0c-ce1cb7251a74">

- [Preview is broken](https://hamlet-git-ham-345-impulsum-vc.vercel.app/ca/san-mateo-county/burlingame#school-rankings)

<img width="1552" alt="Screen Shot 2023-07-12 at 11 37 22" src="https://github.com/cmpsr/composer/assets/1816129/e23f8046-5dd0-4130-8250-483d383deadb">

Reply:
- Debugging it locally, the datapoints are set as `string` [as mentioned above](https://github.com/cmpsr/composer/pull/751#discussion_r1260875212).

<img width="1440" alt="Screen Shot 2023-07-12 at 11 42 32" src="https://github.com/cmpsr/composer/assets/1816129/6fce9624-9505-4ad8-88f1-2d137db70005">